### PR TITLE
Refactored small part of code according to standard quality rules

### PR DESCRIPTION
**Pull request by MaximoOliveira**
Made minor changes that impact the code readability and quality according to rules from https://rules.sonarsource.com/

Code smells with Minor, Major and Critical severity were found.

This is about 10% of the total code smells found and fixed

Based on the feedback from this PR I will determine if I continue fixing the rest of the code!

Primitives should not be boxed just for "String" conversion:
https://rules.sonarsource.com/java/RSPEC-2131

Constructors should not be used to instantiate "String", "BigInteger", "BigDecimal" and primitive-wrapper classes:
https://jira.sonarsource.com/browse/RSPEC-2129

Collapsible "if" statements should be merged:
https://jira.sonarsource.com/browse/RSPEC-1066

Strings should not be concatenated using '+' in a loop:
https://rules.sonarsource.com/java/RSPEC-1643

Local variables should not be declared and then immediately returned or thrown:
https://rules.sonarsource.com/java/RSPEC-1488?search=returned

"@Override" should be used on overriding and implementing methods:
https://rules.sonarsource.com/java/RSPEC-1161?search=override

The diamond operator ("<>") should be used:
https://rules.sonarsource.com/java/RSPEC-2293?search=the%20diamond%20operator

### CODE REVIEW

File: src/main/java/fr/inria/astor/core/faultlocalization/cocospoon/CocoFaultLocalization.java [hunk 1]
```diff
@@ -99,16 +99,14 @@ public FaultLocalizationResult searchSuspicious(ProjectRepairFacade project, Lis
 		}
 		int maxSuspCandidates = ConfigurationProperties.getPropertyInt("maxsuspcandidates");
 
-		if (!ConfigurationProperties.getPropertyBool("considerzerovaluesusp")) {
+		if (Boolean.FALSE.equals(ConfigurationProperties.getPropertyBool("considerzerovaluesusp"))) {
 			candidates.removeIf(susp -> (susp.getSuspiciousValue() == 0));
 		}
 		// We select the best X candidates.
-		int max = (candidates.size() < maxSuspCandidates) ? candidates.size() : maxSuspCandidates;
+		int max = Math.min(candidates.size(), maxSuspCandidates);
```
Comment (MaximoOliveira) [inline src/main/java/fr/inria/astor/core/faultlocalization/cocospoon/CocoFaultLocalization.java:106]: Simplification of the expression

File: src/main/java/fr/inria/astor/core/manipulation/synthesis/dynamoth/combinations/BinaryExpressionEvaluator.java [hunk 2]
```diff
@@ -52,9 +52,7 @@ private Object perform() {
 					throw new RuntimeException("Not comparable");
 				}
 			} else if (secondExpression instanceof Boolean) {
-				if (!(firstExpression instanceof Boolean)) {
```
Comment (MaximoOliveira) [inline src/main/java/fr/inria/astor/core/manipulation/synthesis/dynamoth/combinations/BinaryExpressionEvaluator.java:55]: Condition !(firstExpression instanceof Boolean) is always true

File: src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
```
Comment (MaximoOliveira) [inline src/main/java/fr/inria/astor/core/solutionsearch/AstorCoreEngine.java:13]: This file can be ignored. Seems like it was a formatting change

**Comment by martinezmatias** [comment]
Dear @MaximoOliveira 
Thanks  a lot for the PR. We appreciate the contribution.
However,  we cannot merge the PR for the following main issues:
1) the PR is too large, it modifies many files which is hard for us to manually inspect and review.
2) We "improve" the code of a file while we modify it for incorporating a new feature or a bug fixing.
3) We are not strict of the used of some conventions dictated by SonarSource (e.g.,  as you mentioned, diamont or inline variables). 

Regards
Matias

**Pull request closed without merging**